NIP-XX: Geocaching (#1943)
nostr-protocol/nips: GC.md

NIP-GC
======

Geocaching Events
-----------------

`draft` `optional`

This NIP defines event kinds for geocaching activities on Nostr, enabling users to create, discover, and log geocaches in a decentralized manner.

## Motivation

Geocaching is a global treasure hunting game where participants use GPS coordinates to hide and seek containers ("geocaches") at specific locations. This NIP brings geocaching to Nostr, allowing for censorship-resistant, decentralized geocache listings and logs without relying on centralized platforms.

## Event Kinds

This NIP defines two new event kinds:

- `kind:37515`: Geocache listing (addressable event)

> - `kind:37516`: Geocache log entry

Reviewer: these should be normal kinds, 7515 and 7516 will work much better

Reviewer 2: I think replaceable makes more sense in this case

Reviewer: why?

Reviewer 2: A few of the tags seem intended to change, particularly status, but maybe also hint, image, t. I agree that the log entries should be immutable.

Reviewer: "status" specifically seems to be useless. a log entry will probably work much better, either by the original poster or by someone else.

if you're interested in a geocache you'll likely click on it to fetch its history, and maybe it was abandoned by the original poster but you see that others have reported going to the place and finding nothing, or something like that, then you know it's abandoned.

I think it might be a good idea to create another kind for "status report" or something like that.

## Geocache Listing (kind:37515)

A geocache listing is an addressable event that describes a hidden geocache. The event is addressable to allow cache owners to update their listings (e.g., to provide maintenance updates or adjust coordinates).

### Content

The `.content` field contains the cache description as plain text.

### Tags

Required tags:

> - `d` (required): Unique identifier for the geocache

Reviewer: this doesn't seem to be necessary, each geocache is unique so their identifier is naturally their event id

Reviewer: ```suggestion
- `d` (required): Unique identifier for the geocache
```

- `terrain` (required): Integer from 1-5 indicating physical difficulty (1=wheelchair accessible, 5=specialized equipment required)
- `size` (required): One of: `micro`, `small`, `regular`, `large`
- `cache-type` (required): One of: `traditional`, `multi`, `mystery`, `earth`, `virtual`, `letterbox`, `event`

Optional tags:
- `hint` (optional): An encrypted hint that can help seekers find the cache. Should be ROT13 encoded by convention
- `image` (optional, repeated): Image URLs related to the cache location or container
- `t` (optional, repeated): Hashtags/categories (e.g., "urban", "forest", "historical")
- `published_at` (optional): Unix timestamp when the cache was first hidden
- `status` (optional): Current cache status - `active` (default), `disabled`, or `archived`
- `relay` (optional but recommended, repeated): Preferred relays for log submissions in order of preference

### Example

```json
{
  "kind": 37515,
  "pubkey": "<cache owner's pubkey>",
  "created_at": 1234567890,
  "content": "A scenic cache along the river trail. Please be mindful of muggles during busy hours.",
  "tags": [

>     ["d", "riverside-mystery-2024"],

Reviewer: ```suggestion
```

    ["name", "Riverside Mystery"],
    ["g", "dp3wfg"],
    ["location", "Riverside Park, Portland, OR"],
    ["difficulty", "2"],
    ["terrain", "3"],
    ["size", "small"],
    ["cache-type", "traditional"],
    ["hint", "Ybbx sbe gur byq bnx gerr"],
    ["image", "https://example.com/cache-area.jpg"],
    ["t", "scenic"],
    ["t", "river"],
    ["published_at", "1704067200"],
    ["status", "active"],
    ["relay", "wss://geocaching.example.com"],
    ["relay", "wss://oregon.nostr.com"]
  ]
}
```

## Geocache Log (kind:37516)

A geocache log represents a user's visit to a geocache. These are regular (non-addressable) events.

### Content

The `.content` field contains the log message as plain text.

### Tags

Required tags:

> - `a` (required): Reference to the geocache being logged using NIP-01 `a` tag format: `["a", "37515:<cache-owner-pubkey>:<d-tag>", "<relay-url>"]`. The relay URL SHOULD be the preferred relay from the cache listing if available

Reviewer: ```suggestion
- `e` (required): Reference to the geocache event being logged using. The relay URL SHOULD be the preferred relay from the cache listing if available
```

- `log-type` (required): One of:
  - `found`: Successfully found the cache
  - `dnf`: Did Not Find despite searching
  - `note`: General comment without searching
  - `maintenance`: Maintenance performed by cache owner
  - `disabled`: Cache temporarily disabled by owner
  - `enabled`: Cache re-enabled by owner
  - `archived`: Cache permanently retired

Optional tags:
- `image` (optional, repeated): Image URLs from the visit
- `g` (optional): Geohash of where the log was created (for privacy, should be less precise than cache location)
- `published_at` (optional): Unix timestamp of the actual visit (may differ from event creation time)

### Example

```json
{

>   "kind": 37516,

Reviewer: ```suggestion
  "kind": 7516,
```

  "pubkey": "<finder's pubkey>",
  "created_at": 1234567890,
  "content": "Beautiful location! Took me 20 minutes to find it. TFTC!",
  "tags": [

>     ["a", "37515:abc123...:riverside-mystery-2024", "wss://geocaching.example.com"],

Reviewer: ```suggestion
    ["e", "fbdf73ac656e2dc7579f8a0787b81e8173c1c816090a50fe994a316001f3a875", "wss://geocaching.example.com"],
```
